crates/bdk/src/wallet/mod.rs

```diff
@@ -1685,52 +1758,62 @@ impl<D> Wallet<D> {
     /// transactions related to your wallet into it.
     ///
     /// [`commit`]: Self::commit
-    pub fn apply_update(&mut self, update: Update) -> Result<(), UpdateError>
+    pub fn apply_update(&mut self, update: Update) -> Result<bool, UpdateNotConnectedError>
     where
-        D: persist::PersistBackend<KeychainKind, ConfirmationTime>,
+        D: PersistBackend<ChangeSet>,
     {
-        let changeset = self.keychain_tracker.apply_update(update)?;
-        self.persist.stage(changeset);
-        Ok(())
+        let mut changeset: ChangeSet = self.chain.apply_update(update.chain)?.into();
+        let (_, derivation_additions) = self
+            .indexed_graph
+            .index
+            .reveal_to_target_multi(&update.keychain);
+        changeset.append(derivation_additions.into());
+        changeset.append(self.indexed_graph.apply_update(update.graph).into());
+
+        let changed = !changeset.is_empty();
+        if changed {
+            self.persist.stage(changeset);
+        }
+        Ok(changed)
     }
 
     /// Commits all curently [`staged`] changed to the persistence backend returning and error when this fails.
     ///
     /// [`staged`]: Self::staged
-    pub fn commit(&mut self) -> Result<(), D::WriteError>
+    pub fn commit(&mut self) -> Result<bool, D::WriteError>
     where
-        D: persist::PersistBackend<KeychainKind, ConfirmationTime>,
+        D: PersistBackend<ChangeSet>,
     {
-        self.persist.commit()
+        self.persist.commit().map(|c| c.is_some())
     }
 
     /// Returns the changes that will be staged with the next call to [`commit`].
     ///
     /// [`commit`]: Self::commit
-    pub fn staged(&self) -> &ChangeSet {
+    pub fn staged(&self) -> &ChangeSet
+    where
+        D: PersistBackend<ChangeSet>,
+    {
         self.persist.staged()
     }
 
     /// Get a reference to the inner [`TxGraph`](bdk_chain::tx_graph::TxGraph).
-    pub fn as_graph(&self) -> &bdk_chain::tx_graph::TxGraph {
-        self.keychain_tracker.graph()
+    pub fn as_graph(&self) -> &TxGraph<ConfirmationTimeAnchor> {
```

Comment:
```suggestion
    pub fn tx_graph(&self) -> &TxGraph<ConfirmationTimeAnchor> {
```

I think `as_` doesn't feel right. `Wallet` is not wrapping a `TxGraph` it just has one you can access. Also it's nice to have `tx_` in front of `graph` since it's not clear what it's a graph of.

Reply:
Changed.